index.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * year Input Plugin
+ *
+ *
+ * A simple select input for years based on a range
+ */
+const validation = require('./lib/validation.js');
+const yearRange = require('./lib/yearRange.js');
+
+
+/**
+ * Single year Input Plugin
+ * @module yearInputPlugin
+ */
+module.exports = {
+  name: 'year',
+  description: 'A simple select input for years based on a range',
+  validation: {
+    yearValidation: validation,
+  },
+  scripts: {
+    yearGenerate: yearRange,
+  },
+  inputs: {
+    year: {
+      validation: {
+        function: 'yearValidation',
+        on: 'change',
+      },
+      label: 'Please choose a year',
+      placeholder: 'year',
+      options: {
+        minYear: '1900',
+        maxYear: '2016',
+      },
+      type: 'select',
+      settings: {
+        empty: true,
+      },
+    },
+  },
+  html: '<label for="{{year.id}}">{{year.label}}</label><select id="{{year.id}}" name="{{year.name}}"><option value="{{year.value}}"  class="year-range"></option></select></label>',
```

Comment:
@Snugug I think generated content in input plugins is def something we need, and this scenario is probably the easiest and most direct way to figure out how to do that.

`selected` can be generated, but the code needs to be added. See: https://github.com/punchcard-cms/input-plugin-select/blob/master/index.js#L48 for how to do `selected`.

Another option might be to _not_ use a separate javascript, but instead put the js into the `html` attribute? Or **better yet**, maybe do it with nunjucks??